TTS/bin/train_tts.py

```diff
@@ -485,22 +496,40 @@ def main(args):  # pylint: disable=redefined-outer-name
     if c.use_speaker_embedding:
         speakers = get_speakers(meta_data_train)
         if args.restore_path:
-            prev_out_path = os.path.dirname(args.restore_path)
-            speaker_mapping = load_speaker_mapping(prev_out_path)
-            assert all([speaker in speaker_mapping
-                        for speaker in speakers]), "As of now you, you cannot " \
-                                                   "introduce new speakers to " \
-                                                   "a previously trained model."
-        else:
+            if c.use_external_speaker_embedding_file: # if restore checkpoint and use External Embedding file
```

Comment:
Can we place this speaker embedding loading part into a function so that we can re-use it for a different model or training run?

Reply:
I'm a little confused. I don't see another place to use this, because this code snippet uses load_speaker_mapping and just checks if there is a file speakers.json in args.restore_path, if it exists it will be loaded, otherwise it tries to load c.external_speaker_embedding_file.
Do you see any other use for this, even though we already have load_speaker_mapping?
If so what part do you want me to put in a function?

Reply:
for instance, if we use a different model and write a different train_new_model.py instead of copying and pasting this part it would be better to import this part as a function? Does it make sense?

Reply:
I understand now ! Makes sense :)